source/base/data_out_base.cc

```diff
@@ -5211,44 +5211,33 @@ namespace DataOutBase
     // when writing, first write out all vector data, then handle the scalar
     // data sets that have been left over
     std::vector<bool> data_set_written(n_data_sets, false);
-    for (unsigned int n_th_vector = 0; n_th_vector < vector_data_ranges.size();
-         ++n_th_vector)
+    for (auto range : vector_data_ranges)
       {
-        AssertThrow(std::get<1>(vector_data_ranges[n_th_vector]) >=
-                      std::get<0>(vector_data_ranges[n_th_vector]),
-                    ExcLowerRange(std::get<1>(vector_data_ranges[n_th_vector]),
-                                  std::get<0>(
-                                    vector_data_ranges[n_th_vector])));
-        AssertThrow(std::get<1>(vector_data_ranges[n_th_vector]) < n_data_sets,
-                    ExcIndexRange(std::get<1>(vector_data_ranges[n_th_vector]),
-                                  0,
-                                  n_data_sets));
-        AssertThrow(std::get<1>(vector_data_ranges[n_th_vector]) + 1 -
-                        std::get<0>(vector_data_ranges[n_th_vector]) <=
-                      3,
+        AssertThrow(std::get<1>(range) >= std::get<0>(range),
+                    ExcLowerRange(std::get<1>(range), std::get<0>(range)));
```

Comment:
I think it would be easier to read if you gave the elements of the tuple proper names using
```
  const unsigned int first_component = std::get<0>(range);
  const unsigned int last_component = std::get<1>(range);
  ...
```
and similarly for the other fields.